src/main/java/com/artgallery/infrastructure/ArtRepositoryJson.java

```diff
@@ -47,10 +47,8 @@ public void addPainting(Painting painting) {
     @Override
     public void updatePainting(Painting painting) {
         List<Painting> paintings = getPaintings();
-        paintings.remove(painting);
-        paintings.add(painting);
-        JSONArray jsonArray = ArtMapper.mapPaintings(paintings);
-        writeToFile(jsonArray);
+        paintings.remove(getPainting(painting.getId(), paintings));
```

Comment:
paintings.remove(painting);

Reply:
to jeszcze inaczej bebechy w tej metodzie:
List<Painting> paintings = getPaintings();
paintings.remove(painting);
     paintings.add(painting);
        JSONArray jsonPainting = ArtMapper.mapPaintings(paintings);
        writeToFile(jsonPainting);